lib/dao.js

```diff
@@ -1659,7 +1669,7 @@ DataAccessObject.find = function find(query, options, cb) {
             });
 
             // FIXME: apply "includes" and other transforms - see allCb below
-            memory.all(modelName, ctx.query, options, cb);
+            memory.all(modelName, query, options, cb);
```

Comment:
Actually we already have a test covering this; please see [here](https://github.com/strongloop/loopback-datasource-juggler/blob/063983ccc9257db2e500eb273a585e6a9dad6652/test/persistence-hooks.suite.js#L102)

Anyway, I just added a similar test to make sure there is no regression... Please see [here](https://github.com/strongloop/loopback-datasource-juggler/pull/898/files#diff-145ab49855b38984ba190919e6e7397dR104)